tests/unit_tests/test_workflow.py

```diff
@@ -841,32 +881,63 @@ def test_optlock():
     optlock.update_with_fdr(precursor_df)
 
     assert optlock.has_target_num_precursors is True
-    assert optlock.first_time_to_reach_target is True
+    assert optlock.previously_calibrated is False
+
     optlock.update()
 
-    assert optlock.first_time_to_reach_target is False
     assert optlock.start_idx == 0
 
 
 def test_optlock_batch_idx():
-    library = TEST_LIBRARY()
+    library = create_test_library()
     optlock = optimization.OptimizationLock(library, TEST_OPTLOCK_CONFIG)
 
     optlock.batch_plan = [[0, 100], [100, 2000], [2000, 8000]]
 
     assert optlock.start_idx == 0
 
-    optlock.increase_batch_idx()
+    optlock.update()
```

Comment:
Is there also a code which tests correct decreasing of batch idx?

Reply:
This test does test the correct decreasing to some degree -- it tests that the start_idx goes to 0 -- but I've added an extra assert statement to test the stop_idx also to make this clearer.